src/ModelContextProtocol/Shared/McpSession.cs

```diff
@@ -20,7 +20,13 @@ internal sealed class McpSession : IDisposable
     private readonly RequestHandlers _requestHandlers;
     private readonly NotificationHandlers _notificationHandlers;
 
+    /// <summary>Collection of requests sent on this session and waiting for responses.</summary>
     private readonly ConcurrentDictionary<RequestId, TaskCompletionSource<IJsonRpcMessage>> _pendingRequests = [];
+    /// <summary>
+    /// Collection of requests received on this session and currently being handled. The value provides a <see cref="CancellationTokenSource"/>
+    /// that can be used to request cancellation of the in-flight handler.
+    /// </summary>
+    private readonly ConcurrentDictionary<RequestId, CancellationTokenSource> s_handlingRequests = new();
```

Comment:
Leftover from when it was static?
```suggestion
    private readonly ConcurrentDictionary<RequestId, CancellationTokenSource> _handlingRequests = new();
```